@xen-orchestra/audit-core/src/index.js

```diff
@@ -0,0 +1,121 @@
+// see https://github.com/babel/babel/issues/8450
+import 'core-js/features/symbol/async-iterator'
+
+import assert from 'assert'
+import hash from 'object-hash'
+import { invert } from 'lodash'
+
+export class Storage {
+  constructor() {
+    this.locked = undefined
+  }
+
+  async lock() {
+    if (this.locked !== undefined) {
+      await this.locked
```

Comment:
This does not work, you will have race conditions in case of more than 2 consumers.